ext/strscan/strscan.c

```diff
@@ -450,6 +450,7 @@ strscan_get_charpos(VALUE self)
     GET_SCANNER(self, p);
 
     substr = rb_funcall(p->str, id_byteslice, 2, INT2FIX(0), LONG2NUM(p->curr));
+    Check_Type(substr, T_STRING);
```

Comment:
Would it make sense to rather use `rb_enc_strlen()`, which takes `char *` directly? It seems like a waste to create a String object just for this.

Reply:
Sounds good! Thank you for letting me know, I didn't know the feature 😅 
How about like this https://github.com/ruby/strscan/pull/20/commits/02bfa2c50fcdb41ed301b2e5867c35e382a2b0a8 ?
Tests passed in Ruby 3.0.0. But I'm a newbie in C, so I'm afraid something wrong 🙏